### PR TITLE
Deprecate the Group Chats for Events Feature

### LINKED ISSUE #2297: Mobile - Deprecate the Group Chats for Events Feature
**Is your feature request related to a problem? Please describe.**

To simplify the UI/UX we have decided to deprecate group chats tied to Events so that the draft GSoC 2024 chat idea just focuses on creating basic expected functionality.

The link to chats was previously removed from the bottom of the app’s screen because of this lack of functionality. All other chat code remains.

**Describe the solution you'd like**

This means that all support for this feature must be removed. This includes removing all of the following related items:

1. Sections of screens that contain references this feature
1. Any screens completely dedicated to this feature
1. Underlying code related to the specific support of this feature
1. Any testing code specific to this feature

All other code related to group chats must remain. We only want the group chat feature related to Events to be removed.

**Describe alternatives you've considered**
- N/A

**Approach to be followed (optional)**
- N/A

**Additional context**
The Figma designs showing the relevant Event Group Chat screens can be found here:
- https://docs.talawa.io/docs/design/ux/ux-design-system

**Other**
Issues related to this issue include:
- https://github.com/PalisadoesFoundation/talawa/issues/2297
- https://github.com/PalisadoesFoundation/talawa-api/issues/1618

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Hasnain01-hub**
**What kind of change does this PR introduce?**
Deprecate group chats tied to Events, the aim is to remove the code for Event chat and all the test code related to it.

**Issue Number:**

Fixes #2297

**Did you add tests for your changes?**
NO

**Snapshots/Videos:**

**If relevant, did you update the documentation?**
YES

**Summary**
Remove the code for Event chat, and all the test code related to it.

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Comment by noman2002** [comment]
@Hasnain01-hub Please fix the failing test.

**Review by Ayush0Chaudhary — COMMENTED** [review]
add docs

**Comment by Hasnain01-hub** [comment]
@Ayush0Chaudhary @noman2002 I have removed the docs related to the Event chat since the functionality has been deprecated. Is that acceptable?

**Comment by palisadoes** [comment]
@Hasnain01-hub is this ready for review? It's been some time that their PR has remained inactive

**Comment by Ayush0Chaudhary** [comment]
🔍 Ignore directive for custom lint rule found. Please remove them and add valid documentation.
>>> File name: lib/services/chat_service.dart

>>> File name: lib/services/org_service.dart

>>> File name: lib/services/task_service.dart

>>> File name: lib/view_model/pre_auth_view_models/login_view_model.dart

>>> File name: lib/view_model/pre_auth_view_models/signup_details_view_model.dart

>>> File name: lib/views/after_auth_screens/chat/chat_list_screen.dart

>>> File name: lib/widgets/event_date_time_tile.dart

>>> File name: lib/widgets/member_name_tile.dart

>>> File name: lib/widgets/pinned_carousel_widget.dart

remove top two lies from these file and write docs

File: talawa-mobile-docs/index.md [hunk 1]
```diff
@@ -39,616 +35,304 @@
...
-
 ##### [add_post_view_model](view_model_after_auth_view_models_add_post_view_models_add_post_view_model/view_model_after_auth_view_models_add_post_view_models_add_post_view_model-library.md)
 
-
-
 ##### [app_localization](utils_app_localization/utils_app_localization-library.md)
 
-
-
 ##### [app_settings_page](views_after_auth_screens_app_settings_app_settings_page/views_after_auth_screens_app_settings_app_settings_page-library.md)
```
Comment (Ayush0Chaudhary) [inline talawa-mobile-docs/index.md:52]: why chhange this?

File: talawa-mobile-docs/index.json
Comment (Ayush0Chaudhary) [inline talawa-mobile-docs/index.json:1]: why soo many addition in this file?

File: talawa-mobile-docs/index.md (hunk 1 above)
Comment (Hasnain01-hub) [reply talawa-mobile-docs/index.md:52]: The file contains references to the event chat and adheres to the Prettier formatter. Should I revert it?

File: talawa-mobile-docs/index.json
Comment (Hasnain01-hub) [reply talawa-mobile-docs/index.json:1]: The file also contains some references to the event chat and adheres to the Prettier formatter. Should I revert it?

**Comment by Hasnain01-hub** [comment]
@Ayush0Chaudhary @noman2002 Apologies for the pull request changes. My local branch is outdated and conflicted with the remote branch, prompting the closure and reopening.

**Comment by Hasnain01-hub** [comment]
@palisadoes @noman2002 @Ayush0Chaudhary I have fixed the failing test cases, please check it.

File: lib/views/after_auth_screens/chat/chat_list_screen.dart [hunk 2]
```diff
@@ -1,20 +1,17 @@
-// ignore_for_file: talawa_api_doc
-// ignore_for_file: talawa_good_doc_comments
-
 import 'package:flutter/material.dart';
 import 'package:talawa/views/after_auth_screens/chat/direct_chats.dart';
-import 'package:talawa/views/after_auth_screens/chat/event_chats.dart';
 import 'package:talawa/views/after_auth_screens/chat/select_contact.dart';
 
-/// ChatPage returns a stateless widget for current user Chat Page which renders
+/// ChatPage returns a stateless widget for current user Chat Page which renders.
```
Comment (literalEval) [inline lib/views/after_auth_screens/chat/chat_list_screen.dart:5]: Write this doc in a better way. Don't break the line abruptly just for the sake of ending the first line.

**Review by literalEval — CHANGES_REQUESTED** [review]

**Comment by palisadoes** [comment]
Please review. This is ready

File: lib/views/after_auth_screens/chat/chat_list_screen.dart [hunk 3]
```diff
@@ -1,20 +1,15 @@
...
-// ignore_for_file: talawa_good_doc_comments
-
 import 'package:flutter/material.dart';
 import 'package:talawa/views/after_auth_screens/chat/direct_chats.dart';
-import 'package:talawa/views/after_auth_screens/chat/event_chats.dart';
 import 'package:talawa/views/after_auth_screens/chat/select_contact.dart';
 
-/// ChatPage returns a stateless widget for current user Chat Page which renders
-/// the list of all the users that the current user has chat with.
+/// A stateless widget for the current user's Chat Page, rendering the list of users with whom the current user has chatted.
```
Comment (literalEval) [inline lib/views/after_auth_screens/chat/chat_list_screen.dart:5]: Any doc or code line is supposed to be in the range of 80 characters or so. If it gets longer, you are supposed to turn to multi-line, as discussed in another PR.

**Review by literalEval — CHANGES_REQUESTED** [review]

File: lib/views/after_auth_screens/chat/chat_list_screen.dart (hunk 3 above)
Comment (Hasnain01-hub) [reply lib/views/after_auth_screens/chat/chat_list_screen.dart]: Sorry for the delay, Please check now.

**Comment by Hasnain01-hub** [comment]
@Ayush0Chaudhary @literalEval please review.

File: lib/views/after_auth_screens/chat/chat_list_screen.dart [hunk 4]
```diff
@@ -1,20 +1,15 @@
...
-// ignore_for_file: talawa_good_doc_comments
-
 import 'package:flutter/material.dart';
 import 'package:talawa/views/after_auth_screens/chat/direct_chats.dart';
-import 'package:talawa/views/after_auth_screens/chat/event_chats.dart';
 import 'package:talawa/views/after_auth_screens/chat/select_contact.dart';
 
-/// ChatPage returns a stateless widget for current user Chat Page which renders
-/// the list of all the users that the current user has chat with.
+/// Renders the list of users with whom the current user has chatted.
```
Comment (literalEval) [inline lib/views/after_auth_screens/chat/chat_list_screen.dart:5]: Please remove "renders the". It's a widget, not a rendering function. Yes the widget will be rendered, but all of the UI is rendered, obviously.

**Review by literalEval — CHANGES_REQUESTED** [review]

File: lib/views/after_auth_screens/chat/chat_list_screen.dart (hunk 4 above)
Comment (Hasnain01-hub) [reply lib/views/after_auth_screens/chat/chat_list_screen.dart]: Please check now!

**Comment by noman2002** [comment]
Any update on this PR? Its open since more than a month. Please get this merged.

**Comment by Hasnain01-hub** [comment]
@noman2002 Apologies, sir. I had my exams, which caused the delay. Please review now.

**Review by noman2002 — APPROVED** [review]
LGTM

**Review by palisadoes — APPROVED** [review]

**Pull request merged**